devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -591,6 +609,20 @@ private void executeIfConfigured(String pluginGroupId, String pluginArtifactId,
                         pluginManager));
     }
 
+    private String getGoalPhaseOrNull(String groupId, String artifactId, String goal) {
+        Plugin plugin = getConfiguredPluginOrNull(groupId, artifactId);
+
+        if (plugin == null) {
+            return null;
+        }
+        for (PluginExecution pluginExecution : plugin.getExecutions()) {
+            if (pluginExecution.getGoals().contains(goal)) {
+                return pluginExecution.getPhase();
```

Comment:
This doesn't seem to return the inherited default phase, e.g. the following will _not_ be picked up:
```java
            <executions>
                <execution>
                    <id>make-index</id>
                    <goals>
                        <!-- phase is 'process-classes by default' -->
                        <goal>jandex</goal>
                    </goals>
                </execution>
            </executions>
```
Checking...

Reply:
I'll push a fix. It ain't pretty but it seems we have to fall back to the default phase by our own.